Update CHANGELOG.md (#17281)
hashicorp/terraform-provider-google: CHANGELOG.md

## 5.17.0 (Unreleased)

NOTES:
* cloudbuildv2: changed underlying actuation engine for `google_cloudbuildv2_connection`, there should be no user-facing impact ([#17222](https://github.com/hashicorp/terraform-provider-google/pull/17222))

DEPRECATIONS:

> * container: deprecated support for `relay_mode` field in `advanced_datapath_observability_config` in favor of `enable_relay` field, `relay_mode` field will be removed in upcoming releases ([#17262](https://github.com/hashicorp/terraform-provider-google/pull/17262))

Reviewer: ```suggestion
* container: deprecated support for `relay_mode` field in `google_container_cluster.monitoring_config.advanced_datapath_observability_config` in favor of `enable_relay` field, `relay_mode` field will be removed in a future major release ([#17262](https://github.com/hashicorp/terraform-provider-google/pull/17262))
```

FEATURES:
* **New Resource:** `google_firebase_app_check_debug_token` ([#17242](https://github.com/hashicorp/terraform-provider-google/pull/17242))

> * **New Resource:** `google_network_security_firewall_endpoint` (beta) ([#17208](https://github.com/hashicorp/terraform-provider-google/pull/17208))

Reviewer: ```suggestion
```

* **New Resource:** `google_clouddeploy_custom_target_type` ([#17254](https://github.com/hashicorp/terraform-provider-google/pull/17254))

> * **New Resource:** `google_network_security_security_profile_group` ([#17263](https://github.com/hashicorp/terraform-provider-google/pull/17263))

Reviewer: Looks like this is beta-only

```suggestion
```

IMPROVEMENTS:

> * cloudasset: allowed overriding the billing project for `data.google_cloud_asset_resources_search_all`

Reviewer: ```suggestion
* cloudasset: allowed overriding the billing project for the `google_cloud_asset_resources_search_all` datasource
```

* clouddeploy: added support for `canary_revision_tags`, `prior_revision_tags`, `stable_revision_tags`, and `stable_cutback_duration` to `google_clouddeploy_delivery_pipeline`
* cloudfunctions: expose `version_id` on `google_cloudfunctions_function` ([#17273](https://github.com/hashicorp/terraform-provider-google/pull/17273))

> * compute: promoted `user_ip_request_headers` field on `google_compute_security_policy` resource to GA ([#17271](https://github.com/hashicorp/terraform-provider-google/pull/17271))

Reviewer: ```suggestion
* compute: promoted `user_ip_request_headers` field on `google_compute_security_policy` resource to GA ([#17271](https://github.com/hashicorp/terraform-provider-google/pull/17271))
* container: added support for `enable_relay` field to `google_container_cluster.monitoring_config.advanced_datapath_observability_config` ([#17262](https://github.com/hashicorp/terraform-provider-google/pull/17262))
```

* eventarc: added support for `http_endpoint.uri` and `network_config.network_attachment` to `google_eventarc_trigger` ([#17237](https://github.com/hashicorp/terraform-provider-google/pull/17237))
* healthcare: added `reject_duplicate_message` field to `google_healthcare_hl7_v2_store ` resource ([#17267](https://github.com/hashicorp/terraform-provider-google/pull/17267))
* identityplatform: added `client`, `permissions`, `monitoring` and `mfa` fields to `google_identity_platform_config` ([#17225](https://github.com/hashicorp/terraform-provider-google/pull/17225))
* notebooks: added `desired_state` field to `google_notebooks_instance` ([#17268](https://github.com/hashicorp/terraform-provider-google/pull/17268))
* vertexai: added `feature_registry_source` field to `google_vertex_ai_feature_online_store_featureview` resource ([#17264](https://github.com/hashicorp/terraform-provider-google/pull/17264))
* workbench: added `desired_state` field to `google_workbench_instance` resource ([#17270](https://github.com/hashicorp/terraform-provider-google/pull/17270))

> * workstations: added support for `disable_ssh` in `google_workstations_workstation_config` (beta) ([#17229](https://github.com/hashicorp/terraform-provider-google/pull/17229))

Reviewer: ```suggestion
```
